owncloudApp/src/main/java/com/owncloud/android/files/FileMenuFilter.java

```diff
@@ -117,22 +117,32 @@ public FileMenuFilter(List<OCFile> targetFiles, Account account, ComponentsGette
      * @param menu Options or context menu to filter.
      */
     public void filter(Menu menu, boolean displaySelectAll, boolean displaySelectInverse,
-                       boolean onlyAvailableOffline, boolean sharedByLinkFiles) {
+                       boolean onlyAvailableOffline, boolean sharedByLinkFiles, boolean hasWritePermission,
+                       boolean hasDeletePermission, boolean hasRenamePermission, boolean hasMovePermission,
+                       boolean hasResharePermission) {
```

Comment:
The `FileMenuFilter` already maintains the list of files we need to filter. We could get the permissions from there instead of adding 5 new parameters, right?

Reply:
Yes, it's true. I removed them and accessed their value directly from `FileMenuFilter` 👍